EIPS/eip-6963.md

```diff
@@ -0,0 +1,298 @@
+
+### Identifiers
+
+A locally unique identifier prevents from conflicts using the same globally unique identifier by using UUID v4.0
+
+A globally unique identifier is used for machine-readable detection of the currently active wallet and it can take different formats, for example:
+
+```text
+# lowercase name
+awesomewallet
+
+# legacy JS variable
+isCoolWallet
+
+# reversed domain
+io.dopewallet.app
+```
+
+### Images/Icons
+
+A uri encoded image was chosen to enable flexibility for multiple protocols for fetching and rendering icons, for example:
+
+```sh
+# svg (data uri)
+data:image/svg+xml,<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="32px" height="32px" viewBox="0 0 32 32"><circle fill="red" cx="16" cy="16" r="12"/></svg>
+# png (data uri)
+data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==
+# png (ipfs uri)
+ipfs://QmZo7gsostaUdrV1peFz4cX6Z2TgriFJJP4VahG5knVm29
+# webp (http uri)
+https://ethereum.org/static/f541df14fca86543040c113725b5bd1a/99bcf/metamask.webp
+```
+
+Additionally the image must be squared with 96x96px minimum resolution. Image format is recommended to be lossless like PNG and WebP or alternatively vectorial like SVG. We strongly discourage lossy formats like JPG/JPEG.
+
+### Events
+
+Previous proposal introduced mechanisms that relied on a single window object that could be overriden by multiple parties.
+
+Therefore using an event-based approach we avoid these race conditions and potential attacks by making the communication available both ways.
+
+To follow the Javascript event name conventions, the names are written in present tense and are prefixed with the standard number (`eip6963`).
+
+## Backwards Compatibility
+
+This EIP doesn't require supplanting `window.ethereum`, so it doesn't directly break existing applications. However, the recommended behavior of eventually supplanting `window.ethereum` would break existing applications that rely on it.
+
+## Reference Implementation
+
+### Wallet Provider
+
+Here is a reference implementation for an injected script by a wallet provider to support this new interface in parallel with the existing pattern.
+
+```typescript
+function onPageLoad() {
+  let provider: EIP1193Provider;
+
+  window.ethereum = provider;
+
+  function announceProvider() {
+    const info: EIP6963ProviderInfo = {
+      walletId: "org.example.wallet",
+      uuid: "350670db-19fa-4704-a166-e52e178b59d2",
+      name: "Example Wallet",
+      icon: "https://wallet.example.org/icon.png",
+    };
+    window.dispatchEvent(
+      new CustomEvent("eip6963:announceProvider", {
+        detail: { info, provider },
+      })
+    );
+  }
+
+  window.addEventListener(
+    "eip6963:requestProvider",
+    (event: EIP6963RequestProviderEvent) => {
+      announceProvider();
+    }
+  );
+
+  announceProvider();
+}
+```
+
+### Ethereum library
```

Comment:
>>What does "Ethereum library" mean in this context? Is it talking about a wallet connection library like Web3 Onboard or Rainbowkit?

>That's correct @asciiman! This could include Web3Modal, Web3React, Web3Onboard, RainbowKit, ConnectKit, web3.js, wagmi.js, ethers.js, viem.js, etc

Maybe "Wallet Consumer" would be better here. It could be any of the libraries above, and it could also conceivably be a dApp handling the selection on it's own.